app/src/main/java/com/kickstarter/ui/activities/compose/search/SearchScreen.kt

```diff
@@ -187,35 +187,70 @@ fun SearchScreen(
     }
     val initialCategoryPillText = stringResource(R.string.fpo_category)
     var categoryPillText = remember { mutableStateOf(initialCategoryPillText) }
+    var currentSort by remember { mutableStateOf(DiscoveryParams.Sort.POPULAR) }
+    var currentCategory by remember { mutableStateOf<Category?>(null) }
+
+    val activeBottomSheet = remember {
+        mutableStateOf("")
+    }
+
+    val categorySheetState = rememberModalBottomSheetState(
+        initialValue = Hidden,
+        skipHalfExpanded = true
+    )
+
+    val sortSheetState = rememberModalBottomSheetState(
+        initialValue = Hidden,
+        skipHalfExpanded = false
+    )
 
     ModalBottomSheetLayout(
-        sheetState = sheetState,
+        sheetState = if (activeBottomSheet.value == "categories") categorySheetState else sortSheetState,
```

Comment:
instead of String comparison with categories you can use now `activeBottomSheet == FilterRowPillType.CATEGORY`